### DISCUSSION TITLE
Uploading and inserting at once?

### DISCUSSION THREAD

**Discussion by riancz** [supabase/supabase#4608] [Questions]
Hi guys, I wonder is it possible to upload to storage and write to database in one request?
For example:
You want to insert a post in your database which has some pictures attached to it. You want to do it in one request and populate a column with links to the pictures.

How I do it now:
First I upload all the pictures (to some "folder" in my bucket), keep all the links and then send an insert to database to which I add the links.

Would be cool to have the pictures saved in folders named as the ID of the post, do it all in one request and not to throw them all to a big pile of photos.

**Comment by GaryAustin1** [comment]
I don't know if this method would work for your environment...
When creating a new "note" in my javascript client, I upload multiple files and store them using LocalForage which handles file objects into clients indexedDB and set filenames in the note object.
Then after I save the note object, if it has filenames, I upload them from LocalForage, upload the files to supabase with the path including the ID of my note object, then delete the LocalForage copy.
I can then on a delete of the note object. trigger a delete of all files with the object's ID in the path, and by not storing the files until I have a note in the database make sure files don't get orphaned if the client never saves the note for some reason.

**Reply by riancz** [reply]
I am not sure if I understand what you are suggesting but it sounds like you still need to upload the files and then insert the row, which is how I do it now.

**Reply by GaryAustin1** [reply]
Yes, but I upload the files from localForage AFTER the insert so I have the id of the row and can use that as part of the path for my files which was last part of your comment.  Plus by uploading the files after the insert they won't get "orphaned" from my note object if the row insert fails for some reason.
There is no operation to do it all at once.
